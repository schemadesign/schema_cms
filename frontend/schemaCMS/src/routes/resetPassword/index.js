export { default as ResetPassword } from './resetPassword.container';
