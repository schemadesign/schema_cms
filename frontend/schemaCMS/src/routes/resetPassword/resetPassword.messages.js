/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  validate: {
    id: 'home.validate',
    defaultMessage: 'Validating...',
  },
});
