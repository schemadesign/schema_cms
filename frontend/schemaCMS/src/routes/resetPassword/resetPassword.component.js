import React, { PureComponent } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import { Container } from './resetPassword.styles';
import messages from './resetPassword.messages';

export class ResetPassword extends PureComponent {
  static propTypes = {
    resetPassword: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.resetPassword();
  }

  render() {
    return (
      <Container>
        <FormattedMessage {...messages.validate} />
      </Container>
    );
  }
}
