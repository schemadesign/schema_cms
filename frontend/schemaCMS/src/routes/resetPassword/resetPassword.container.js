import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose } from 'ramda';

import { ResetPassword } from './resetPassword.component';
import { UserAuthActions } from '../../modules/userAuth';

const mapStateToProps = createStructuredSelector({});

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      resetPassword: UserAuthActions.resetPassword,
    },
    dispatch
  );

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter
)(ResetPassword);
