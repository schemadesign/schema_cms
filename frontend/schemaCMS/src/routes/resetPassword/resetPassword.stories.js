import React from 'react';
import { storiesOf } from '@storybook/react';

import { ResetPassword } from './resetPassword.component';

const defaultProps = {
  resetPassword: Function.prototype,
};

storiesOf('ResetPassword', module).add('Default', () => <ResetPassword {...defaultProps} />);
