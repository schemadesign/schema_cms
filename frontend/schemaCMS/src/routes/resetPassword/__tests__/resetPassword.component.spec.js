import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';

import { ResetPassword } from '../resetPassword.component';

describe('ResetPassword: Component', () => {
  const defaultProps = {
    resetPassword: Function.prototype,
  };

  const component = props => <ResetPassword {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should call resetPassword function on componentDidMount', () => {
    const resetPassword = spy();
    render({ resetPassword });

    expect(resetPassword).to.have.been.calledOnce;
  });
});
