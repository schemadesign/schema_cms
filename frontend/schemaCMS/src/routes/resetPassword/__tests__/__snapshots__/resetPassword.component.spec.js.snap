// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword: Component should render correctly 1`] = `
<styled.div>
  <FormattedMessage
    defaultMessage="Validating..."
    id="home.validate"
    values={Object {}}
  />
</styled.div>
`;
